packages/parser/src/parser/parse.ts

```diff
@@ -73,3 +78,58 @@ export function* tokenizeContent(content: string): IterableIterator<Token> {
     log.error('Error occured while getting token: %o', e)
   }
 }
+
+/**
+ * Parses a content string to extract its value and any parsing errors.
+ * It doesn't support values that are asynchronously parsed
+ *
+ * @param content The content string to parse.
+ * @param functions An optional set of helper functions used during parsing.
+ * @param filename An optional name of the file from which the content was read, used for error context.
+ * @returns An object containing:
+ *          - `value`: The parsed value of the content.
+ *          - `errors`: A list of errors encountered during parsing.
+ */
+
+export const parseValue = ({
+  content,
+  functions = {},
+  filename = '',
+}: {
+  content: string
+  functions?: Functions
+  filename?: string
+}): {
+  value: Value
+  errors: ParseError[]
+} => {
+  const context: ParseContext = {
+    calcSourceMap: false,
+    filename,
+    functions,
+    lexer: new PeekableLexer(content),
+    errors: [],
+    listTypes: {},
+    mapTypes: {},
+    sourceMap: new SourceMap(),
+    valuePromiseWatchers: [],
+  }
+  let result: ConsumerReturnType<Value> | undefined
+  try {
+    result = consumeValue(context)
+  } catch (e) {
+    processParseError(context, e)
+  }
+  if (context.valuePromiseWatchers.length > 0) {
```

Comment:
since we saw that for `myFunc("some.png")`, `valuePromiseWatchers`is empty, and `value` is a `Promise`:
```suggestion
  if (context.valuePromiseWatchers.length > 0 || isPromise(result?.value)) {
```
and I would also add a test for that.